### ISSUE TITLE
portfinder@1.0.22 returns random port (Docker examples)

### ISSUE THREAD

**Issue by danlynn** [http-party/node-portfinder#87]
Something changed that is causing portfinder to return random ports instead of the requested port which is open:
```
docker run --rm -it node:10.16.3 bash
root@0ba5c01217f7:/# npm install portfinder@1.0.22
npm WARN saveError ENOENT: no such file or directory, open '/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/package.json'
npm WARN !invalid#1 No description
npm WARN !invalid#1 No repository field.
npm WARN !invalid#1 No README data
npm WARN !invalid#1 No license field.

+ portfinder@1.0.22
added 6 packages from 6 contributors and audited 6 packages in 0.999s
found 0 vulnerabilities

root@0ba5c01217f7:/# node
> const PortFinder = require('portfinder');
undefined
> PortFinder.getPort({port: 4200, host: 'localhost'}, function(err, port) {console.debug(`err = ${err}, port = ${port}`)});
undefined
> err = null, port = 21663

> PortFinder.getPort({port: 4200, host: 'localhost'}, function(err, port) {console.debug(`err = ${err}, port = ${port}`)});
undefined
> err = null, port = 26449
```

Here is the same code running successfully with portfinder@1.0.21:
```
docker run --rm -it node:10.16.3 bash
root@55537e7efc22:/# npm install portfinder@1.0.21
npm WARN saveError ENOENT: no such file or directory, open '/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/package.json'
npm WARN !invalid#1 No description
npm WARN !invalid#1 No repository field.
npm WARN !invalid#1 No README data
npm WARN !invalid#1 No license field.

+ portfinder@1.0.21
added 6 packages from 6 contributors and audited 6 packages in 1.023s
found 0 vulnerabilities

root@55537e7efc22:/# node
> const PortFinder = require('portfinder');
undefined
> PortFinder.getPort({port: 4200, host: 'localhost'}, function(err, port) {console.debug(`err = ${err}, port = ${port}`)});
undefined
> err = null, port = 4200

> PortFinder.getPort({port: 4200, host: 'localhost'}, function(err, port) {console.debug(`err = ${err}, port = ${port}`)});
undefined
> err = null, port = 4200
```

**Comment by fenghuijiuzhuan** [comment]
+1

**Comment by okbeng03** [comment]
+1

**Comment by SmithXiong** [comment]
npm install portfinder@1.0.21 revert is ok

**Comment by jelhan** [comment]
This is a duplicate of #84. It should be fixed by #86.

**Comment by eriktrom** [comment]
this should be fixed, i rolled back v1.0.23 -> v1.0.21
